metagraph/plugins/numpy/algorithms.py

```diff
@@ -119,3 +125,12 @@ def np_nodemap_reduce(x: NumpyNodeMap, func: Callable[[Any, Any], Any]) -> Any:
     if not isinstance(func, np.ufunc):
         func = np.frompyfunc(func, 2, 1)
     return func.reduce(present_values)
+
+
+@concrete_algorithm("util.node_embedding.apply")
+def np_embedding_apply(
+    embedding: NumpyNodeEmbedding, nodes: NumpyVector
+) -> NumpyMatrix:
+    indices = embedding.nodes[nodes.value]
+    matrix = embedding.matrix.np_matrix(copy=False)[indices]
```

Comment:
Given the matrix must always be dense, this technically isn't necessary and could be written as `embedding.matrix.value[indices]`.

Reply:
https://github.com/metagraph-dev/metagraph/pull/133/commits/eb0eda1c191b9170b685a88b55e1b3319b42d464 refactors `np_embedding_apply` to implement this suggestion.